src/Symfony/Component/Security/Guard/Authenticator/AbstractFormLoginAuthenticator.php

```diff
@@ -71,9 +75,11 @@ public function onAuthenticationSuccess(Request $request, TokenInterface $token,
     {
         // if the user hit a secure page and start() was called, this was
         // the URL they were on, and probably where you want to redirect to
-        $targetPath = $request->getSession()->get('_security.'.$providerKey.'.target_path');
+        if ($request->getSession() instanceof SessionInterface) {
+            $targetPath = $request->getSession()->get('_security.'.$providerKey.'.target_path');
+        }
 
-        if (!$targetPath) {
+        if (!isset($targetPath) || !$targetPath) {
```

Comment:
I would prefer to explicitly initialize `$targetPath` to `null`above and make a check on `null` here.